src/NuGet.Clients/NuGet.CommandLine/Common/CommandLineResponseFile.cs

```diff
@@ -0,0 +1,154 @@
+                    {
+
+                        if (ParseArgsResponseFileRecursionDepth > MaxRecursionDepth)
+                        {
+                            throw new ArgumentException(string.Format(LocalizedResourceManager.GetString("Error_ResponseFileMaxRecursionDepth"), MaxRecursionDepth));
+                        }
+
+                        ParseArgsResponseFileRecursionDepth++;
+                        var nestedResponseFileArgs = ParseArgsResponseFiles(new string[] { a });
+                        if (nestedResponseFileArgs.Length != 0)
+                        {
+                            parsedArgs.AddRange(nestedResponseFileArgs);
+                        }
+
+                        continue;
+                    }
+
+                    var parsedResponseFileArg = a.Trim();
+                    if (!string.IsNullOrWhiteSpace(parsedResponseFileArg))
+                    {
+                        parsedArgs.Add(parsedResponseFileArg);
+                    }
+                }
+            }
+
+            return parsedArgs.ToArray();
+        }
+
+        /// <summary>
+        /// Splits up a string of command line arguments into an array
+        /// </summary>
+        /// <param name="unsplitArguments">The string to split</param>
+        /// <returns>An array with split command line arguments</returns>
+        private static string[] SplitArgs(string unsplitArguments)
+        {
+            if (string.IsNullOrWhiteSpace(unsplitArguments))
+            {
+                return new string[0];
+            }
+
+            var ptrToSplitArgs = CommandLineToArgvW(unsplitArguments, out int numberOfArgs);
+            if (ptrToSplitArgs == IntPtr.Zero)
+            {
+                return new string[0];
+            }
+
+            try
+            {
+                var splitArgs = new string[numberOfArgs];
+                for (var i = 0; i < numberOfArgs; i++)
+                {
+                    splitArgs[i] = Marshal.PtrToStringUni(Marshal.ReadIntPtr(ptrToSplitArgs, i * IntPtr.Size));
+                }
+
+                return splitArgs;
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(ptrToSplitArgs);
+            }
+        }
+
+        /// <summary>
+        /// Parses a Unicode command line string and returns an array of pointers to the command line arguments, 
+        /// along with a count of such arguments, in a way that is similar to the standard C run-time argv and argc values.
+        /// </summary>
+        /// <param name="commandLine">Unicode string that contains the full command line.</param>
+        /// <param name="argsLength">Pointer to an int that receives the number of array elements returned</param>
+        /// <returns></returns>
+        [DllImport("shell32.dll", SetLastError = true)]
+        static extern IntPtr CommandLineToArgvW([MarshalAs(UnmanagedType.LPWStr)] string commandLine, out int argsLength);
```

Comment:
is this going to work on mono? is there a reason we need to use native code to do the parsing? i'd rather use managed code if it's an option at all.

Reply:
Probably not, I should put this behind preprocessor directive.

Reply:
Here is the equivalent for unix: https://linux.die.net/man/3/wordexp. Looks like DllImport using CoreCLR also works on that platform. Calling the low level command line parser is the right thing to do here rather than a re-write in managed.

Reply:
ifdef response file support so only desktop is compiled in

Reply:
Fixed

Reply:
Fixed